### PR TITLE
Added low mach preconditioner

### PR DESCRIPTION
**Pull request by NasserFlexCompute**

### CODE REVIEW

File: flow360/component/flow360_params/solvers.py [hunk 1]
```diff
@@ -144,6 +144,13 @@ class NavierStokesSolver(GenericFlowSolverSettings):
     linear_solver:
         Linear solver settings
 
+    low_mach_preconditioner:
+        Uses preconditioning for accelerating low Mach number flows.
+
+    low_mach_preconditioner_threshold:
+        For flow regions with mach numbers smaller than threshold, the input Mach number to the preconditioner is
```
Comment (chai-flex) [inline flow360/component/flow360_params/solvers.py:151]: M should be capital here

File: flow360/component/flow360_params/solvers.py [hunk 2]
```diff
@@ -174,6 +181,22 @@ class NavierStokesSolver(GenericFlowSolverSettings):
...
+    )
+
+    # pylint: disable=arguments-differ,invalid-name
+    def to_solver(self, params, **kwargs) -> NavierStokesSolver:
+        """
+        Set preconditioner threshold to freestream Mach number
+        """
+
+        if self.low_mach_preconditioner:
+            self.low_mach_preconditioner_threshold = params.freestream.Mach
```
Comment (chai-flex) [inline flow360/component/flow360_params/solvers.py:196]: What if Mach in freestream is 0? Do we use MachRef

File: flow360/component/flow360_params/validations.py [hunk 3]
```diff
@@ -474,3 +474,36 @@ def _check_numerical_dissipation_factor_output(values):
...
+    navier_stokes_solver = values.get("navier_stokes_solver")
+    if navier_stokes_solver is not None and not isinstance(
+        navier_stokes_solver, IncompressibleNavierStokesSolver
+    ):
+        low_mach_preconditioner = navier_stokes_solver.low_mach_preconditioner
+        if not low_mach_preconditioner and "lowMachPreconditionerSensor" in _get_all_output_fields(
+            values
+        ):
+            raise ValueError(
+                "Low-Mach preconditioner output requested but low-Mach preconditioner mode is not enabled."
```
Comment (chai-flex) [inline flow360/component/flow360_params/validations.py:489]: comma after requested

File: flow360/component/flow360_params/solvers.py (hunk 2 above)
Comment (NasserFlexCompute) [reply flow360/component/flow360_params/solvers.py:196]: No. There's a minimum value specified in the solver. (0.005)

**Comment by chai-flex** [comment]
Needs lint fix

**Review by chai-flex — APPROVED** [review]

**Pull request merged**